sdcm/nemesis.py

```diff
@@ -2031,27 +2020,54 @@ def _add_drop_column(self, drop=True, add=True):  # pylint: disable=too-many-bra
         if not added_columns_info['column_names']:
-            drop = False
+            drop = []
         if drop:
             drop = self._add_drop_column_generate_columns_to_drop(added_columns_info)
         if add:
             add = self._add_drop_column_generate_columns_to_add(added_columns_info)
         if not add and not drop:
             return
-        # TBD: Scylla does not support DROP and ADD in the same statement
-        if drop:
-            cmd = f"ALTER TABLE {self._add_drop_column_target_table[1]} DROP ( {', '.join(drop)} );"
-            if self._add_drop_column_run_cql_query(cmd, self._add_drop_column_target_table[0]):
-                for column_name in drop:
-                    column_type = added_columns_info['column_names'][column_name]
-                    del added_columns_info['column_names'][column_name]
-        if add:
-            cmd = f"ALTER TABLE {self._add_drop_column_target_table[1]} " \
-                  f"ADD ( {', '.join(['%s %s' % (col[0], col[1]) for col in add])} );"
-            if self._add_drop_column_run_cql_query(cmd, self._add_drop_column_target_table[0]):
-                for column_name, column_type in add:
-                    added_columns_info['column_names'][column_name] = column_type
-                    column_type.remember_variant(added_columns_info['column_types'])
+
+        def remove_columns_callback(columns, result):
+            self.log.debug(f"remove_columns_info {columns}, {result}")
+            for column_name in columns:
+                del added_columns_info['column_names'][column_name]
+
+        def add_columns_callback(columns, result):
+            self.log.debug(f"add_columns_info {columns}, {result}")
+            for column_name, column_type in columns:
+                added_columns_info['column_names'][column_name] = column_type
+                column_type.remember_variant(added_columns_info['column_types'])
+
+        with self.cluster.cql_connection_patient(self.target_node) as session:
+            session.default_consistency_level = ConsistencyLevel.ALL
+            cmds = []
+            table_name = f"{self._add_drop_column_target_table[0]}.{self._add_drop_column_target_table[1]}"
+            # TBD: Scylla does not support DROP and ADD in the same statement
+            if bool(random.getrandbits(1)):  # randomize commands by splitting 1 add/drop command into multiple
+                if drop:
+                    cmds.append({"callback": partial(remove_columns_callback, columns=drop),
+                                 "cmd": f"ALTER TABLE {table_name} DROP ( {', '.join(drop)} );"})
+                if add:
+                    cmds.append({"callback": partial(add_columns_callback, columns=add),
+                                 "cmd": f"ALTER TABLE {table_name} ADD "
+                                        f"( {', '.join(['%s %s' % (col[0], col[1]) for col in add])} );"})
+            else:
+                for column in drop:
```

Comment:
Where do we control the number of columns that's appended ?

Reply:
in self._add_drop_column_columns_info  as I see